ml_task_runner.py

```diff
@@ -0,0 +1,135 @@
+import json
+import os
+import signal
+import sys
+import time
+
+import backoff
+import nbformat
+import nbconvert
+
+from api_clients.task import TaskClient
+from api_clients.core import CoreClient
+
+class MLTaskRunner(object):
+    shutting_down = False
+    download_complete = False
+
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self.task_client = TaskClient(configuration['services']['task-service']['base_url'],
+                                      configuration['auth_token'],
+                                      configuration['services']['task-service']['worker_id'])
+        self.core_client = CoreClient(configuration['services']['core-service']['base_url'],
+                                      configuration['auth_token'],
+                                      configuration['services']['task-service']['worker_id'])
+        self.task = None
+
+    @staticmethod
+    def run_notebook(notebook_name, base_path='notebooks/'):
+        notebook_path = os.path.join(os.getcwd(), base_path, notebook_name + '.ipynb')
```

Comment:
I've been using pathlib lately (builtin to Python 3.4+) and it makes working with paths much nicer. Consider, if not here, for the future.